idx/memory/memory.go

```diff
@@ -242,17 +237,17 @@ func (m *MemoryIdx) add(def *schema.MetricDefinition) error {
 		log.Debug("memory-idx: creating branch %s with child %s", branch, nodes[i])
```

Comment:
After this loop `startPos` is not used anymore. So instead of declaring `i` and assigning the value of `startPos` to it, we could also just use `startPos` in place of `i` here